packages/integral_isolates/lib/src/integral_isolates.dart

```diff
@@ -1,236 +1,15 @@
-    final isolateConfiguration = IsolateConfiguration(
-      callback,
-      message,
-      debugLabel,
-      flow.id,
-    );
-
-    backpressureStrategy.add(MapEntry(completer, isolateConfiguration));
-    _handleIsolateCall();
-    return completer.future;
-  }
-
-  /// Closes down the isolate and cancels all jobs currently in queue.
-  ///
-  /// This function should always be called when you are done with the isolate
-  /// to not leak memory and isolates.
-  ///
-  /// After this function is called, you cannot continue using the isolate.
-  @mustCallSuper
-  Future dispose() async {
-    // TODO(lohnn): prevent user from adding more work to the isolate after this function is called.
-    _closePort?.send('close');
-    _isolateToMainPort.cancel();
-    backpressureStrategy.dispose();
-  }
-
-  /// If the worker is currently running, this bool will be set to true
-  bool _isRunning = false;
-
-  Future _handleIsolateCall() async {
-    if (_initCompleter == null) {
-      throw InitException();
-    } else if (!_initCompleter!.isCompleted) {
-      await _initCompleter!.future;
-    }
-    if (!_isRunning && backpressureStrategy.hasNext()) {
-      _isRunning = true;
-      final configuration = backpressureStrategy.takeNext();
-
-      try {
-        _mainToIsolatePort.send(configuration.value);
-
-        final response = await _isolateToMainPort.next;
-        if (response is _SuccessIsolateResponse) {
-          configuration.key.complete(response.response);
-        } else if (response is _ErrorIsolateResponse) {
-          configuration.key.completeError(response.error, response.stackTrace);
-        } else {
-          // TODO(lohnn): Should not be possible? Notify the developer of issue?
-        }
-      } catch (e, stackTrace) {
-        configuration.key.completeError(e, stackTrace);
-      }
-      _isRunning = false;
-      _handleIsolateCall();
-    }
-  }
-}
-
-Future _isolate(SendPort isolateToMainPort) async {
-  final mainToIsolateStream = ReceivePort();
-  final closePort = ReceivePort();
-
-  isolateToMainPort.send(
-    _IsolateSetupResponse(
-      mainToIsolateStream.sendPort,
-      closePort.sendPort,
-    ),
-  );
-
-  closePort.first.then((_) {
-    mainToIsolateStream.close();
-    closePort.close();
-  });
-
-  await for (final data in mainToIsolateStream) {
-    try {
-      if (data is IsolateConfiguration) {
-        try {
-          isolateToMainPort.send(
-            _IsolateResponse.success(
-              data.flowId,
-              await data.applyAndTime(),
-            ),
-          );
-        } catch (error, stackTrace) {
-          isolateToMainPort.send(
-            _IsolateResponse.error(data.flowId, error, stackTrace),
-          );
-        }
-      } else {
-        isolateToMainPort.send(null);
-      }
-    } catch (_) {
-      isolateToMainPort.send(null);
-    }
-  }
+  // ignore: public_member_api_docs
+  @Deprecated('Use StatefulIsolate directly instead')
```

Comment:
I think it is enough to deprecate the class if I remember correctly

Reply:
It's good to say which version it will be removed in too, but I don't think that matters when you are sub-v1